Error building dso-slam with Visual Studio

I get an error building dso-slam with Visual Studio 2015. The first error message is as follows:

>Error	C2976	'std::deque': too few template arguments	dso	C:\XXX\dso-slam\dso\src\FullSystem\FullSystem.h	310	


Line 310 of FullSystem.h:
`std::deque<FrameHessian*> unmappedTrackedFrames;`

Is this related to the the definition of `struct FrameHessian` in HessianBlacks.h?

Actually, I'd like to fix this issue but I don't know where to start searching. Maybe there is someone more experienced that may give a hint on this. What would be useful information to look at?

Kind regards

PS: And why does boost come into play? May this be a hint for an incompatibility to my current boost version (1.63)?
>C:\XXX\dso-slam\dso\src\FullSystem/FullSystem.h(310): error C2976: 'std::deque': too few template arguments
>C:\libs\boost\v1.63.0\boost/detail/container_fwd.hpp(131): note: see declaration of 'std::deque'

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -31,6 +31,10 @@ set(CMAKE_CXX_FLAGS
 #   "${SSE_FLAGS} -O3 -g -std=c++0x -fno-omit-frame-pointer"
 )
 
+if (MSVC)
+     set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /EHsc")
+endif (MSVC)
+
 # Sources files
 set(dso_SOURCE_FILES
   ${PROJECT_SOURCE_DIR}/src/FullSystem/FullSystem.cpp

--- a/src/FullSystem/FullSystem.h
+++ b/src/FullSystem/FullSystem.h
@@ -25,6 +25,7 @@
 #pragma once
 #define MAX_ACTIVE_FRAMES 100
 
+#include <deque>
 #include "util/NumType.h"
 #include "util/globalCalib.h"
 #include "vector"

